### PR TITLE
CV3-69 Refactor filter events

### PR DESCRIPTION
**Pull request by mifeille**
#### Description
There are two filter functions in the event model i.e filter_events_by_date_range and filter_events_by_room. These two functions have lots of repetition. They can be merged into one.

#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

Checklist
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations

#### How to Test
Test if `allEvents` and `allEventsByRoom` queries are still working as expected.

#### JIRA
[CV3-69](https://andela-apprenticeship.atlassian.net/browse/CV3-69)

### CODE REVIEW

File: api/events/models.py [hunk 1]
```diff
@@ -31,54 +31,40 @@ class Events(Base, Utility):
     app_booking = Column(Boolean, nullable=True, default=False)
 
 
-def filter_events_by_date_range(query, start_date, end_date):
+def filter_event(start_date, end_date, room_id=None):
     """
-    Returns events that fall within the date range supplied
+    Returns filtered events if the room id
+    or start date and end date are supplied,
+    and return all events when those parameters are not supplied.
```
Comment (adafia) [inline api/events/models.py]: It looks good, please reword the last line of this docstring. It suggests functionality contrary to what the function does.

**Review by adafia — COMMENTED** [review]
The filter event by room returns the same response ("Events do not exist for the date range") no matter the date range I provide. Is this the expected behaviour?

**Review by MCFrank16 — APPROVED** [review]

**Review by MCFrank16 — APPROVED** [review]

**Comment by mifeille** [comment]
> The filter event by room returns the same response ("Events do not exist for the date range") no matter the date range I provide. Is this the expected behaviour?

have you checked if you have events for the date range you are providing in your database? if yes, can you please send me the queries you are using so that I can see where the problem is coming from.

**Comment by adafia** [comment]
The function takes in a calendar Id so I was under the impression that the events will be queried from a remote calendar from the Google Calendar API. Nevertheless, my date range is pretty wide-spanning: November 2017 to November 2019.

**Comment by mifeille** [comment]
> The function takes in a calendar Id so I was under the impression that the events will be queried from a remote calendar from the Google Calendar API. Nevertheless, my date range is pretty wide-spanning: November 2017 to November 2019.

Please use one of the calendar ids in your rooms table.

**Review by adafia — APPROVED** [review]

**Review by joshuaocero — APPROVED** [review]

**Pull request merged**